crates/common/tedge_utils/src/file.rs

```diff
@@ -0,0 +1,112 @@
+use nix::unistd::*;
+use std::fs::File;
+use std::os::linux::fs::MetadataExt;
+use std::os::unix::fs::PermissionsExt;
+use std::{fs, io};
+use users::{get_group_by_name, get_user_by_name};
+
+pub fn create_directory_with_user_group(
+    grp_user: &str,
+    dirs: Vec<&str>,
+) -> Result<(), anyhow::Error> {
+    for directory in dirs {
```

Comment:
Why a vector of directory?

This function should focus on creating a directory. The loop must be outside.

```suggestion
    directory: Vec<&str>,
) -> Result<(), anyhow::Error> {
```

Reply:
Thought of having one call with all the directories. But, now I fixed it.